plugins/modules/idrac_firmware_info.py

```diff
@@ -104,34 +106,55 @@
   }
 '''
 
-
 import json
 from ansible_collections.dellemc.openmanage.plugins.module_utils.dellemc_idrac import iDRACConnection, idrac_auth_params
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import remove_key
 from ansible.module_utils.basic import AnsibleModule
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
-from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
+from urllib.error import URLError, HTTPError
+
+GET_IDRAC_FIRMWARE_DETAILS_URI = "/redfish/v1/UpdateService/Oem/Dell/DellSoftwareInventory?$select=Members"
+GET_IDRAC_FIRMWARE_URI = "/redfish/v1/UpdateService/Oem/Dell/DellSoftwareInventory"
+
+
+def get_idrac_firmware_info(idrac, module):
+    try:
+        response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_URI)
+
+        if response.status_code == 200:
+            details_response = idrac.invoke_request(method='GET', uri=GET_IDRAC_FIRMWARE_DETAILS_URI)
+
+            if details_response and details_response.status_code == 200 and details_response.json_data:
+                filtered_data = remove_key(details_response.json_data)
+                return filtered_data
+    except Exception as e:
+        with iDRACConnection(module.params) as idrac:
+            firmware_details = idrac.update_mgr.InstalledFirmware
+            return firmware_details
 
 
-# Main
 def main():
     specs = {}
     specs.update(idrac_auth_params)
     module = AnsibleModule(
         argument_spec=specs,
-        supports_check_mode=True)
+        supports_check_mode=True
+    )
 
     try:
-        with iDRACConnection(module.params) as idrac:
-            msg = idrac.update_mgr.InstalledFirmware
+        with iDRACRedfishAPI(module.params) as idrac:
+            firmware_info = get_idrac_firmware_info(idrac, module)
     except HTTPError as err:
         module.fail_json(msg=str(err), error_info=json.load(err))
     except URLError as err:
-        module.exit_json(msg=str(err), unreachable=True)
+        module.fail_json(msg=str(err), unreachable=True)
```

Comment:
Avoid fail_json and remove unused import

Reply:
Done